### PR TITLE
fix(e2e): stabilize flaky HuggingFace LLM assertions

### PR DESCRIPTION
**Pull request by jlost**
**What this PR does / why we need it**:

The four Qwen2 chat/completion e2e tests assert exact LLM output (`== "The result of 2 + 2 is 4."`), which is non-deterministic on CPU CI runners. The chat streaming test (`test_huggingface_openai_chat_completions_streaming`) fails intermittently.

**Root cause**: The exact source of non-determinism in CI has not been identified. Vanilla PyTorch on CPU is fully deterministic (verified: 50 identical forward passes with varying thread counts produce bit-identical logits). The CI container includes Intel Extension for PyTorch (IPEX) and multi-threaded OpenMP, which may introduce floating-point variance, but this has not been confirmed.

What IS confirmed: the old prompt has narrow logit margins at several token positions. When generating the 4th word of the response, the model scores `" "` (leading to the correct answer) and `" adding"` (leading to a wrong answer) only 2.875 points apart. If any source of variance flips the winner at that position, the model produces an entirely different sentence (e.g. `"The result of adding two numbers is always their sum."`).

**Fix** (three parts):

1. **Chat tests** (instruct model): Change prompt from `"What is the result of 2 + 2?"` to `"2+2="` with system message `"Answer with just the number, nothing else."`. The new prompt produces only one token (`"4"`) with a minimum logit gap of 7.5 — over 3x wider than the old prompt's weakest point. Verified 100% deterministic across 150 runs at temperatures 0.5–1.0 with sampling enabled.

2. **Completion tests** (base model): Change prompt from `"What is the result of 2 + 2?"` to a few-shot pattern (`"1+1=2\n3+3=6\n5+5=10\n2+2="`). The old prompt has a minimum logit gap of only 1.25 on the base model. The few-shot pattern produces `"4"` as the very first token with a 4.5 logit gap. Verified 100% reliable across 50 runs at temperatures 0.3–0.7.

3. **All four tests**: Relax assertions from exact string match to `"4" in text`. Streaming tests additionally verify that SSE chunks were received (`len(chunks) > 0`), confirming the streaming protocol works end-to-end.

**Root cause analysis**:

Token-by-token logit gap comparison (greedy decoding):

Old prompt (`"What is the result of 2 + 2?"`):
```
Step        Token   Top1 logit   Top2 logit        Gap   Top2 token
   0        'The'      23.3750      21.0000     2.3750          '2'
   1    ' result'      26.3750      23.2500     3.1250       ' sum'
   2        ' of'      29.1250      25.3750     3.7500        ' is'
   3          ' '      24.8750      22.0000     2.8750    ' adding'  ← flake point
   4          '2'      29.0000      19.5000     9.5000          '1'
  ...
```

New prompt (`"2+2="` with constrained system message):
```
Step        Token   Top1 logit   Top2 logit        Gap   Top2 token
   0          '4'      23.8750      15.4375     8.4375          '0'
   1        <EOS>      22.6250      15.1250     7.5000          '.'
```

Regardless of where the variance originates, the new prompt is far less susceptible: its minimum gap (7.5) is 3x the old prompt's weakest point (2.375), and it generates only 1 token vs 12.

**Type of changes**

- [x] Bug fix (non-breaking change which fixes an issue)

**Feature/Issue validation/testing**:

- [x] Verified new prompt produces deterministic output across 150+ runs with sampling at temperatures 0.5–1.0
- [x] Verified streaming chunk assertions catch empty/broken SSE responses
- [x] Analyzed per-token logit gaps to confirm fix reduces susceptibility to variance

**Checklist**:

- [x] Have you added unit/e2e tests that prove your fix is effective or that this feature works?
- [x] Has code been commented, particularly in hard-to-understand areas?
- [x] Have you made corresponding changes to the documentation?

**Special notes for your reviewer**:

The chat and completion tests use different prompt strategies because the models behave differently. The instruct model responds well to a constrained system message (`"2+2="`), but the base model treats bare `"2+2="` as a math exercise and generates Chinese homework at higher temperatures. The few-shot pattern works for the base model by establishing an arithmetic format the model continues.

```release-note
NONE
```

### CODE REVIEW

**Comment by andresllh** [comment]
great catch @jlost!
looks good on my end!
